ambry-router/src/test/java/com.github.ambry.router/ChunkFillTest.java

```diff
@@ -194,8 +197,11 @@ private void fillChunksAndAssertSuccess()
     random.nextBytes(putContent);
     final ReadableStreamChannel putChannel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(putContent));
     FutureResult<String> futureResult = new FutureResult<String>();
+    MockTime time = new MockTime();
+    MockNetworkClientFactory networkClientFactory = new MockNetworkClientFactory(vProps, null, 0, 0, 0, null, time);
     PutOperation op = new PutOperation(routerConfig, routerMetrics, mockClusterMap, responseHandler, putBlobProperties,
-        putUserMetadata, putChannel, futureResult, null, new MockTime());
+        putUserMetadata, putChannel, futureResult, null,
+        new ReadyForPollCallback(networkClientFactory.getNetworkClient()), null, time);
     numChunks = op.getNumDataChunks();
     compositeBuffers = new ByteBuffer[numChunks];
     final AtomicReference<Exception> operationException = new AtomicReference<Exception>(null);
```

Comment:
Do we have a test that ensures that a ChunkFiller thread shuts down when notified and PutManager is closed?